### DISCUSSION TITLE
Post Digit deployment clarification/questions

### DISCUSSION THREAD

**Discussion by pradeepkumarcm-egov** [upyog/UPYOG#3] [Q&A]
1. How to setup "developers working environment" from local laptop -> git repo-> ci/cd -> AWS dev deployment.
2. Git code maintenance process (how a developer can check in? what all repo? and how we can trigger deployment?), how team structure can be maintained in git etc. 
3. Post-deployment configuration (document is not clear)
4. MDMS sample data preparation and configuration session for test ULB in Kerala.

**Comment by subhashini-egov** [comment]
> 1. How to setup "developers working environment" from local laptop -> git repo-> ci/cd -> AWS dev deployment.
> Before setting up a developer's working environment, the following are prerequisites:

- DIGIT deployed using installer
- GitHub code is cloned onto the local laptop.
- CD/CI pipeline setup --> https://docs.digit.org/urban/platform/setup-digit/jenkins-setup

Docker or other registries are setup for docker images to be pushed.
Once above has been done, a developer can start setting up their development environment following this doc --> https://core.digit.org/developer-guide
In the GitHub repo, branches should be created for dev, uat, qa. Each developer should branch for small, incremental development and merge into one of dev, qa or uat as appropriate. For a detailed read of Git branching strategies, please read this: https://www.gitkraken.com/learn/git/best-practices/git-branch-strategy

> 2. Git code maintenance process (how a developer can check in? what all repo? and how we can trigger deployment?), how team structure can be maintained in git etc.

See above answer for Git processes. Most of the work will happen in the forked DIGIT-OSS repo. Once Jenkins is set up, various build jobs for different modules for  will be enabled/shown in the build page. Commit the code to a branch, build the module from Jenkins. Instructions here --> https://docs.digit.org/urban/platform/setup-digit/jenkins-setup/ci-cd-build-job-pipeline-setup. Deployment is also covered here.

**Comment by pradeepkumarcm-egov** [comment]
3. Post-deployment configurations are not clear 

       Refer: https://docs.digit.org/urban/platform/setup-digit/install-on-cloud#3.-post-deployment-steps
          Seed data required for PGR Module are provided here. The end user will load the same using postman scripts. Seed data includes localization key/values, employee and citizen users creation script, user role mapping, access token generation, and pgr workflow configuration scripts.
          For other services, we need to refer to configuration documents defined for each service. The MDMS, configuration, and seed data required for each module are defined in the service config document

      Eg: Refer Obps module: https://docs.digit.org/product/modules/online-building-plan-approval-system-obpas/obpas-service-configuration
            TL Module: https://docs.digit.org/product/modules/trade-license-tl/tl-service-configuration


    4. MDMS sample data preparation and configuration session for test ULB in Kerala.
 
        The master data setup document present in https://docs.digit.org/urban/platform/configure-digit/setting-up-master-data 

        Modulewise, the master data configuration document is present in https://docs.digit.org/urban/platform/configure-digit/configuring-master-data-templates/module-setup. 

        The state-level masters are documented in https://docs.digit.org/urban/platform/configure-digit/configuring-master-data-templates/environment-setup
